### ISSUE TITLE
FR/Question: shared id as snapp descriptor

### ISSUE THREAD

**Issue by ITwrx** [loki-project/loki-network#1337]
I'm not sure if this is already possible, or if this issue would be a FR. 

Is it possible to have a shared ID as a snapp descriptor in addition to the snapp's unique ID/descriptor, so that all copies of a distributed snapp could find each other without a central openDHT server/snapp?
thanks

**Comment by majestrate** [comment]
I don't understand the question, can you rephrase? Specifically what is a shared ID in this case? Do you mean 1 .loki address across many lokinet clients?

**Comment by ITwrx** [comment]
well i don't know if it would be an *address*, but some same identifier i could set for each snapp of a certain type, so that my distributed individual snapps could be located as part of my custom "app group" while still maintaining their ip address/location anonymity. so, i thought it would be an additional custom descriptor so that we could set a shared_id and use the existing ability of the loki networks LNS (loki name service IIRC) to have custom virtual groups/psuedo sub networks of snapps. i want my snapps (of a certain app/type) to be able to find each other using loki without an additional openDHT server if possible. sorry i can't describe better as i don't know how this stuff works on a low level very well.

**Comment by majestrate** [comment]
we recently added (in the `dev` branch) the ability to have .loki addresses provide many SRV records, which could allow you to point to any number of other .loki addresses. requires your application to query with dns for SRV records but I think that is pretty close to what you are trying to do.

**Comment by ITwrx** [comment]
sounds like it might work. thanks

i'm not sure yet, but i'm thinking i might like to make a desktop app that has an http client and an API backend built in that both use the loki network in the background, while providing a desktop app experience for the end user. Each snapp instance needs to be able to be a client to other snapp instances' APIs and API server for other snapp instances' clients on the loki network.  So instead of a centralized snapp serving many clients through a standard browser, each copy would be it's own server, client and gui node in a decentralized and anonymized pseudo network of snapps that work together to provide needed functionality. I was trying to figure out if this could be done with the current state of the loki ecosystem. If it ends up being too much trouble (on my end), i may stick to a more traditional web app structure, but ones that can talk to each other.

p.s. i'll leave y'all to decide when to close this issue.
